### PR TITLE
change marshal indent to 2

### PR DESCRIPTION
**Pull request by natasha41575**
change kyaml/yaml.Marshal to use 2 spaces for indent instead of 4

This may affect users of kyaml libraries. Is that okay? I'm guessing if they really need this they can use yaml.Marshal directly.

### CODE REVIEW

File: kyaml/fn/runtime/runtimeutil/runtimeutil_test.go [hunk 1]
```diff
@@ -1150,16 +1150,16 @@ metadata:
 `,
 			expectedFn: `
 container:
-    image: foo:v1.0.0
-    mounts:
-      - type: bind
```
Comment (Shell32-Natsu) [inline kyaml/fn/runtime/runtimeutil/runtimeutil_test.go:1155]: Why these list items have 2 space identations?

Comment (natasha41575) [reply kyaml/fn/runtime/runtimeutil/runtimeutil_test.go:1155]: The indent was 4, and I think '- ' counts as 2 spaces for the sake of indentation.

**Review by KnVerey — COMMENTED** [review]
I'm in favour of making `Encoder` and `Marshal` consistent and I seriously doubt the difference was intentional. If this affected Kustomize output, we couldn't do it (at least not without a major version bump, and we probably wouldn't want to at all), but curiously it doesn't seem to based on the tests. Re: library consumers, it would be great to get a 👍 on this from KPT folks too.

File: kyaml/yaml/alias.go [hunk 2]
```diff
@@ -21,7 +22,16 @@ type Style = yaml.Style
...
-var Marshal = yaml.Marshal
+var Marshal = func (in interface{}) (out []byte, err error) {
+	var buf bytes.Buffer
+	e := NewEncoder(&buf)
+	e.SetIndent(2)
+	err = e.Encode(in)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(buf.String()), nil
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply kyaml/yaml/alias.go:33]: Yes. You can use buf.Bytes() instead.

**Comment by phanimarupaka** [comment]
> I'm in favour of making `Encoder` and `Marshal` consistent and I seriously doubt the difference was intentional. If this affected Kustomize output, we couldn't do it (at least not without a major version bump, and we probably wouldn't want to at all), but curiously it doesn't seem to based on the tests. Re: library consumers, it would be great to get a 👍 on this from KPT folks too.

This is a good change. kpt had this problem with writing results file and I fixed it [using encoder](https://github.com/GoogleContainerTools/kpt/pull/2174/files#diff-614beac5384ae24bb857a6505c25bb2cad2f88cc36126fc8d4e1975e042ae862) assuming current marshal behavior is intentional. So it's a 👍 from kpt team.

Will this be a surprise to kustomize users ? I think few workflows in kustomize use encoder and few use marshal.

File: kyaml/yaml/alias.go [hunk 3]
```diff
@@ -21,7 +22,16 @@ type Style = yaml.Style
 type TypeError = yaml.TypeError
 type Unmarshaler = yaml.Unmarshaler
 
-var Marshal = yaml.Marshal
+var Marshal = func (in interface{}) (out []byte, err error) {
+	var buf bytes.Buffer
+	e := NewEncoder(&buf)
+	e.SetIndent(2)
```
[the remark this answers is not in the document]
Comment (natasha41575) [reply kyaml/yaml/alias.go]: done

File: kyaml/yaml/alias.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (natasha41575) [reply kyaml/yaml/alias.go]: done

**Comment by natasha41575** [comment]
> Will this be a surprise to kustomize users ? I think few workflows in kustomize use encoder and few use marshal.

I did a quick check and it looks like most places kustomize that use kyaml.Marshal only so that they can construct another struct from the result. For writing to output, it generally uses kio.ByteWriter which uses `Encoder`, and also based on tests I don't think it'll have an impact on kustomize.

**Comment by KnVerey** [comment]
> > Will this be a surprise to kustomize users ? I think few workflows in kustomize use encoder and few use marshal.

> I did a quick check and it looks like most places kustomize use kyaml.Marshal only so that they can construct another struct from the result. For writing to output, it generally uses kio.ByteWriter which uses Encoder, and also based on tests I don't think it'll have an impact on kustomize.

Thanks for checking the code. I'm inclined to agree. If it did impact Kustomize output, we couldn't make this change, but I think we can be confident that it doesn't. We know from the yaml.v3 bump PR that many, many tests do fail if the indent changes, and the only ones we had to fix here are calling Marshal themselves.

/lgtm

**Pull request merged**